mutable_tree.go

```diff
@@ -138,12 +156,99 @@ func (tree *MutableTree) Import(version int64) (*Importer, error) {
 	return newImporter(tree, version)
 }
 
+// Iterate iterates over all keys of the tree. The keys and values must not be modified,
+// since they may point to data stored within IAVL. Returns true if stopped by callnack, false otherwise
+func (t *MutableTree) Iterate(fn func(key []byte, value []byte) bool) (stopped bool) {
+	if t.root == nil {
+		return false
+	}
+
+	if t.isFastCacheEnabled() {
```

Comment:
should we flip the conditions here for simplicity? (if !t.isFastCacheEnabled() { // do simple thing })

Thanks for dealing with this complexity 🙏